build/Dockerfile

```diff
@@ -86,7 +86,13 @@ RUN --mount=type=secret,id=STEAM_USERNAME \
         linux32/steamconsole.so \
         update_hosts_cached.vdf \
         /root/.steam \
-        /root/Steam
+        # Do not remove /root/Steam/appcache. If `/root/Steam/appcache` is removed, `steamcmd` re-downloads `hlds/*` games  from scratch whenever the game is updated.
+        # /root/Steam/appcache \
```

Comment:
If this only applies to `hlds`, perhaps its better to remove the directory only for `hlds`. Looks like the `appcache` directory does take up quite a bit of space.

```
$ docker run -it local/hl2mp:6630498 'du -hs ~/Steam'
5.4M    /root/Steam
```

Reply:
The directory doesn't appear to add much size with each update with no changes in game version but might do so for actual game updates.

```
docker history local/hl2mp:6630498-layered 
IMAGE          CREATED          CREATED BY                                      SIZE      COMMENT
45e89cd9b7ee   11 seconds ago   RUN |3 INSTALL_COUNT=1 STEAM_LOGIN= CACHE_KE…   12.6MB    buildkit.dockerfile.v0
<missing>      11 seconds ago   ARG CACHE_KEY                                   0B        buildkit.dockerfile.v0
<missing>      11 seconds ago   ARG STEAM_LOGIN                                 0B        buildkit.dockerfile.v0
<missing>      11 seconds ago   ARG INSTALL_COUNT=1                             0B        buildkit.dockerfile.v0
<missing>      17 minutes ago   RUN |3 INSTALL_COUNT=1 STEAM_LOGIN= CACHE_KE…   8.07MB    buildkit.dockerfile.v0
<missing>      17 minutes ago   ARG CACHE_KEY                                   0B        buildkit.dockerfile.v0
<missing>      17 minutes ago   ARG STEAM_LOGIN                                 0B        buildkit.dockerfile.v0
<missing>      17 minutes ago   ARG INSTALL_COUNT=1                             0B        buildkit.dockerfile.v0
<missing>      18 minutes ago   RUN |3 INSTALL_COUNT=1 STEAM_LOGIN= CACHE_KE…   8.07MB    buildkit.dockerfile.v0
<missing>      18 minutes ago   ARG CACHE_KEY                                   0B        buildkit.dockerfile.v0
<missing>      18 minutes ago   ARG STEAM_LOGIN                                 0B        buildkit.dockerfile.v0
<missing>      18 minutes ago   ARG INSTALL_COUNT=1                             0B        buildkit.dockerfile.v0
<missing>      19 minutes ago   RUN |3 INSTALL_COUNT=1 STEAM_LOGIN= CACHE_KE…   12.6MB    buildkit.dockerfile.v0
<missing>      19 minutes ago   ARG CACHE_KEY                                   0B        buildkit.dockerfile.v0
<missing>      19 minutes ago   ARG STEAM_LOGIN                                 0B        buildkit.dockerfile.v0
<missing>      19 minutes ago   ARG INSTALL_COUNT=1                             0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   CMD [""]                                        0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ENTRYPOINT ["/bin/bash" "-c"]                   0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   WORKDIR /server                                 0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ENV PATH=/usr/local/sbin:/usr/local/bin:/usr…   0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ENV SERVER_DIR=/server                          0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ENV CLIENT_APPID=320                            0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ENV MOD=                                        0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ENV APPID=232370                                0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ENV LD_LIBRARY_PATH=.:bin:                      0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   RUN |9 SERVER_DIR=/server APPMANIFEST_AR_URL…   4B        buildkit.dockerfile.v0
<missing>      26 minutes ago   RUN |9 SERVER_DIR=/server APPMANIFEST_AR_URL…   742MB     buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG CACHE_KEY                                   0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG STEAM_LOGIN                                 0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG INSTALL_COUNT                               0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG CLIENT_APPID                                0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG FIX_APPMANIFEST=false                       0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG MOD                                         0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG APPID                                       0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG APPMANIFEST_AR_URL=https://github.com/st…   0B        buildkit.dockerfile.v0
<missing>      26 minutes ago   ARG SERVER_DIR=/server                          0B        buildkit.dockerfile.v0
<missing>      4 years ago      /bin/sh -c #(nop)  LABEL game_distributor=st…   0B        
<missing>      4 years ago      /bin/sh -c #(nop)  ENV PATH=/usr/local/sbin:…   0B        
<missing>      4 years ago      /bin/sh -c #(nop)  ENV STEAMCMD_DIR=/steamcmd   0B        
<missing>      4 years ago      /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8 LANG…   0B        
<missing>      4 years ago      |3 DEBIAN_FRONTEND=noninteractive STEAMCMD_A…   151MB     
<missing>      4 years ago      /bin/sh -c #(nop) WORKDIR /steamcmd             0B        
<missing>      4 years ago      /bin/sh -c #(nop)  ARG STEAMCMD_AR_URL=https…   0B        
<missing>      4 years ago      /bin/sh -c #(nop)  ARG STEAMCMD_DIR=/steamcmd   0B        
<missing>      4 years ago      /bin/sh -c #(nop)  ARG DEBIAN_FRONTEND=nonin…   0B        
<missing>      4 years ago      /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B        
<missing>      4 years ago      /bin/sh -c mkdir -p /run/systemd && echo 'do…   7B        
<missing>      4 years ago      /bin/sh -c rm -rf /var/lib/apt/lists/*          0B        
<missing>      4 years ago      /bin/sh -c set -xe   && echo '#!/bin/sh' > /…   745B      
<missing>      4 years ago      /bin/sh -c #(nop) ADD file:01a5c4f2b1dcc8f8a…   116MB 
```

Reply:
Ok I've tweaked the steps. https://github.com/startersclan/docker-sourceservers/pull/113/commits/a3fade19ea7abcf76dfe37cf736f10df76916a8e @leojonathanoh

Reply:
results lgm on my branch :https://github.com/joeltimothyoh/docker-sourceservers/pull/13/commits/a3fade19ea7abcf76dfe37cf736f10df76916a8e @leojonathanoh

Reply:
yes it applies only to `hlds`.

Reply:
it appears that there's no diff between `build` and 1 `update`:


```sh 
$ container-diff diff hldsdev/cstrike:latest hldsdev/cstrike:1127

-----Size-----

Image size difference between hldsdev/cstrike:latest and hldsdev/cstrike:1127: None
```

<details>

```sh
$ docker history hldsdev/cstrike:latest 
IMAGE          CREATED                  CREATED BY                                      SIZE      COMMENT
520657836038   Less than a second ago   RUN |4 INSTALL_COUNT=1 STEAM_LOGIN=true GAME…   0B        buildkit.dockerfile.v0
<missing>      1 second ago             RUN |4 INSTALL_COUNT=1 STEAM_LOGIN=true GAME…   8.8MB     buildkit.dockerfile.v0
<missing>      1 second ago             ARG GAME                                        0B        buildkit.dockerfile.v0
<missing>      1 second ago             ARG GAME_VERSION                                0B        buildkit.dockerfile.v0
<missing>      1 second ago             ARG STEAM_LOGIN                                 0B        buildkit.dockerfile.v0
<missing>      1 second ago             ARG INSTALL_COUNT=1                             0B        buildkit.dockerfile.v0
<missing>      45 hours ago             CMD [""]                                        0B        buildkit.dockerfile.v0
<missing>      45 hours ago             ENTRYPOINT ["/bin/bash" "-c"]                   0B        buildkit.dockerfile.v0
<missing>      45 hours ago             RUN |10 SERVER_DIR=/server APPMANIFEST_AR_UR…   0B        buildkit.dockerfile.v0
<missing>      46 hours ago             WORKDIR /server                                 0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV PATH=/usr/local/sbin:/usr/local/bin:/usr…   0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV SERVER_DIR=/server                          0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV GAME=cstrike                                0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV GAME_VERSION=1127                           0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV CLIENT_APPID=10                             0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV MOD=cstrike                                 0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV APPID=90                                    0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ENV LD_LIBRARY_PATH=.:bin:                      0B        buildkit.dockerfile.v0
<missing>      46 hours ago             RUN |10 SERVER_DIR=/server APPMANIFEST_AR_UR…   3B        buildkit.dockerfile.v0
<missing>      46 hours ago             RUN |10 SERVER_DIR=/server APPMANIFEST_AR_UR…   832MB     buildkit.dockerfile.v0
<missing>      46 hours ago             ARG GAME                                        0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG GAME_VERSION                                0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG STEAM_LOGIN                                 0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG INSTALL_COUNT=                              0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG CLIENT_APPID                                0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG FIX_APPMANIFEST=false                       0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG MOD                                         0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG APPID                                       0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG APPMANIFEST_AR_URL=https://github.com/st…   0B        buildkit.dockerfile.v0
<missing>      46 hours ago             ARG SERVER_DIR=/server                          0B        buildkit.dockerfile.v0
<missing>      4 years ago              /bin/sh -c #(nop)  LABEL game_distributor=st…   0B        
<missing>      4 years ago              /bin/sh -c #(nop)  ENV PATH=/usr/local/sbin:…   0B        
<missing>      4 years ago              /bin/sh -c #(nop)  ENV STEAMCMD_DIR=/steamcmd   0B        
<missing>      4 years ago              /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8 LANG…   0B        
<missing>      4 years ago              |3 DEBIAN_FRONTEND=noninteractive STEAMCMD_A…   151MB     
<missing>      4 years ago              /bin/sh -c #(nop) WORKDIR /steamcmd             0B        
<missing>      4 years ago              /bin/sh -c #(nop)  ARG STEAMCMD_AR_URL=https…   0B        
<missing>      4 years ago              /bin/sh -c #(nop)  ARG STEAMCMD_DIR=/steamcmd   0B        
<missing>      4 years ago              /bin/sh -c #(nop)  ARG DEBIAN_FRONTEND=nonin…   0B        
<missing>      4 years ago              /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B        
<missing>      4 years ago              /bin/sh -c mkdir -p /run/systemd && echo 'do…   7B        
<missing>      4 years ago              /bin/sh -c rm -rf /var/lib/apt/lists/*          0B        
<missing>      4 years ago              /bin/sh -c set -xe   && echo '#!/bin/sh' > /…   745B      
<missing>      4 years ago              /bin/sh -c #(nop) ADD file:01a5c4f2b1dcc8f8a…   116MB  
```

</details>

So the growth has to do with modified dates changing etc.

Reply:
that's good to know.

Reply:
continue discussion in https://github.com/startersclan/docker-sourceservers/pull/113#discussion_r1125613897